### DISCUSSION TITLE
How is Rustdesk "pro" legal or ethical? Isn't creating proprietary software against the community?

### DISCUSSION THREAD

**Discussion by Darin755** [rustdesk/rustdesk#5942] [Q&A] [answered]
So Rustdesk and the rustdesk server are licensed under agpl. That license is a copy left license that it designed to protect the community from people wishing to take the communities work to make proprietary software. 

However, it seems that the Rustdesk pro version lacks such protections. It looks like it may even be proprietary which is against the community. I wouldn't have a problem with it if the source code was given to you when you purchase it but it seems that the source code is locked away. This will lead to vendor lockin and antifeatures as their isn't a way for vendors to see, modify or distribute the source code. 

Am I the only one disturbed by Rustdesk developing proprietary software?

**Comment by SergeyMy** [comment]
The Rustdesk client is completely open source. also
RustDesk ID/rendezvous server
RustDesk relay server

**Comment by open-trade** [comment, marked as answer]
### First:
If you understand the history of RustDesk Server, you'll know that RustDesk Server was initially closed source. Later, parts of it were made open source, in v1.1.6 (https://github.com/rustdesk/rustdesk-server/releases/tag/1.1.6). Therefore, RustDesk Server Pro is not derived from the open-source RustDesk Server (OSS). Instead, the open-source RustDesk Server (OSS) was derived from the closed-source RustDesk Server Pro.

If you do not believe, you can check this https://github.com/rustdesk/rustdesk-server/releases/tag/1.1.5
https://github.com/rustdesk/rustdesk-server/archive/refs/tags/1.1.5.zip contains only readme, no source code.

### Second:

> Since they are the owners of the rustdesk-server program they can do whatever they want with it. In principle the owner of a piece of software can issue licenses however they please. This includes mutiple licenses (all the OSS repos licensed under MIT/Apache 2.0 are an example). So they could simply issue a more permissive license for themselves.
